pytorch_lightning/trainer/trainer.py

```diff
@@ -2094,7 +2094,11 @@ def _results(self) -> Optional[ResultCollection]:
             return active_loop._results
 
     def _exit_gracefully_on_signal(self) -> None:
-        if _fault_tolerant_training() and self._terminate_gracefully:
+        if _fault_tolerant_training():
+            # the signal should be sent to rank 0
```

Comment:
are you sure? if the signal does not get sent to all processes, only some of them will enter this broadcast call and the rest will hang (they will never run the _exit_gracefully_on_signal method).

Reply:
This code works only if the signal is sent to global rank 0. I didn't manage to make it work otherwise. This would need more investigation.